vaadin-upload-flow-parent/vaadin-upload-testbench/src/main/java/com/vaadin/flow/component/upload/testbench/UploadElement.java

```diff
@@ -147,20 +145,15 @@ public void removeFile(int index) {
      *            the number of seconds to wait for the upload to finish
      */
     private void waitForUploads(int maxSeconds) {
-        Timeouts timeouts = getDriver().manage().timeouts();
-        timeouts.setScriptTimeout(maxSeconds, TimeUnit.SECONDS);
-
         String script = "var callback = arguments[arguments.length - 1];"
-                + "var upload = arguments[0];" + "let intervalId;"
-                + "intervalId = window.setInterval(function() {"
-                + "  var inProgress = upload.files.filter(function(file) { return file.uploading;}).length >0;"
-                + "  if (!inProgress) { " //
-                + "    window.clearInterval(intervalId);" //
-                + "    callback();" //
-                + "  }" //
-                + "}, 500);";
-        getCommandExecutor().getDriver().executeAsyncScript(script, this);
-
+                + "var upload = arguments[0];" //
+                + "var inProgress = upload.files.filter(function(file) { return file.uploading;}).length >0;"
+                + "callback(!inProgress);";
+
+        waitUntil(
+                driver -> (Boolean) getCommandExecutor().getDriver()
+                        .executeAsyncScript(script, UploadElement.this),
```

Comment:
I guess this can be transformed to executeScript now that the script is synchronous.

Reply:
fixed